src/sklearn_evaluation/plot/precision_recall.py

```diff
@@ -2,152 +2,372 @@
+    plot_metrics_avg_precision.append(micro_avg_precision)
+
+    for i in range(n_classes):
+        precision, recall, avg_precision = _precision_recall_metrics(
+            y_true_bin[:, i], y_score[:, i]
+        )
+        plot_metrics_recall.append(recall.tolist())
+        plot_metrics_precision.append(precision.tolist())
+        plot_metrics_avg_precision.append(avg_precision)
+
+    return plot_metrics_recall, plot_metrics_precision, plot_metrics_avg_precision
+
+
+def _plot_metrics_multiclass(recall, precision, labels, ax):
+    for r, p, l in zip(recall, precision, labels):
+        ax.plot(r, p, label=l)
+    return ax
+
+
+def _plot_metrics_binary(recall, precision, label, ax):
+    ax.plot(recall, precision, label=label)
+    return ax
+
+
+class PrecisionRecall(AbstractPlot):
     """
-    Plot precision-recall curve.
+    Plot precision recall curve.
 
     Parameters
     ----------
-    y_true : array-like, shape = [n_samples]
-        Correct target values (ground truth).
-    y_score : array-like, shape = [n_samples] or [n_samples, 2] for binary
-              classification or [n_samples, n_classes] for multiclass
+    recall : array-like, shape = [n_samples], when task is binary classification.
+             or shape = [n_classes, n_samples], when task is multiclass classification.
+
+    precision : array-like, shape = [n_samples], when task is binary classification,
+                or shape = [n_classes, n_samples], when task is multiclass
+                classification.
+
+    average_precision : float, when task is binary classification,
+                    or list of floats, when task is multiclass classification
+
+    label : string when task is binary classification, optional
+            list of strings when task is multiclass classification
+            this is used for labelling the curves. Defaults to precision recall.
+            Make sure that the order of the labels corresponds to the order in
+            which recall/precision arrays are passed to the constructor.
+            If label is not passed for multiclass classification, default labels are
+            generated : ['Micro-average', 'Class 1', 'Class 2', ... ]
 
-        Target scores (estimator predictions).
     ax : matplotlib Axes
         Axes object to draw the plot onto, otherwise uses current Axes
 
+    Examples
+    --------
+    .. plot:: ../examples/precision_recall_oop.py
+
+    .. plot:: ../examples/precision_recall_add.py
+
     Notes
     -----
-    It is assumed that the y_score parameter columns are in order. For example,
-    if ``y_true = [2, 2, 1, 0, 0, 1, 2]``, then the first column in y_score
-    must contain the scores for class 0, second column for class 1 and so on.
+    .. versionadded:: 0.10.0
+    """
 
+    @modify_exceptions
+    @SKLearnEvaluationLogger.log(feature="plot", action="precision-recall-init")
+    def __init__(self, recall, precision, average_precision, label=None):
```

Comment:
Let's swap the order recall <> precision to be consistent with the plot's name.

I'm confused about why we still need the average precision parameter.

The idea is that the constructor can take data in multiple formats. Simplest one would be:

```
recall = [X, Y, Z]
precision = [I, J, K]
label = "some label"
```

however, for the multi-class, we have a list of lists:

```
recall = [[X, Y, Z], [X2, Y2, Z2]]
precision = [[I, J, K], [I2, J2, K2]]
label = ["some label", "another label"]
```
Since labels are arbitrary, we could pass "micro-average" there and eliminate the average precision. This is how the new ROC implementation works.

> I have resolved the review comments. On the discussion about labels , I checked the latest ROC code, it looks the code is considering the first value as Micr-average. 

I think ROC uses the label to determine where is the micro-average, it doesn't use the first value. right, @yafimvo? 

> My concern was the case where user calls constructor directly and doesn't pass the labels parameter. In this case how will default labels be generated. For now I have mentioned this in the docstring for the user's reference.

In such case, we can just define generic labels such as PR curve 1, PR curve 2

Reply:
@edublancas Yes. the ROC constructor accepts `fpr`, `tpr` and `labels` (optional), then the `ROC.plot()` generates a curve for each `fpr[i]`, `tpr[i]`, and `labels[i]`. 

The old API - `plot.roc(y_true, y_score)` calls `ROC.from_raw_data(y_true, y_score)` and as a default the 1st index is the micro-average curve. 
We can add a flag to control this (something like: `ROC.from_raw_data(y_true, y_score, include_micro_average=False)` )

Reply:
> I'm confused about why we still need the average precision parameter.

This is the method definition of the `average_precision_score`
```
def average_precision_score(
    y_true, y_score, *, average="macro", pos_label=1, sample_weight=None
):
```

For the case where user is calling the constructor directly with recall, precision metrics (and not y_true, y_score) , how will we calculate the micro-average precision score?